### PR TITLE
2021 07 05/media files relative links

### PR DESCRIPTION
**Pull request by chachasikes**
Adding:
* Relative link rewriting for pages & posts, body and og meta content
* Variables for the production & development hosts
* Variable for controlling if the relative links should be overwritten
* `content` data in api endpoint, it's needed for components

### CODE REVIEW

**Comment by chachasikes** [comment]
For drought build, still needs ability to access the media files (in wordpress/media) - this will look for them in a folder called 'files' that needs to be added to the build.

The data probably needs something similar... maybe 'data' - we can append & update the function after 11ty version works.

**Comment by chachasikes** [comment]
Also, @jbum  and I renamed the local settings file @aaronhans  - to not conflict with other settings, and also to not be same name as the folder.

**Comment by chachasikes** [comment]
And to make it point outside the repo.

**Comment by chachasikes** [comment]
Variables for development not activated - might not be necessary... if you duplicated the endpoints.

File: WordpressSync/debug.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 //Loading environment variables
-const { Values } = require('../local.settings.json');
+const { Values } = require('../../local-wordpress-to-github.settings.json');
```
Comment (carterm) [inline WordpressSync/debug.js:2]: the path for `local.settings.json` is the default for the Azure functions SDK.  This process is a way to have side-by-side compatibility with the file that Azure functions uses already.

File: WordpressSync/endpoints.schema.json [hunk 2]
```diff
@@ -91,6 +115,10 @@
                       "wordpress/pages"
                     ]
                   },
+                  "RelativePaths": {
```
Comment (carterm) [inline WordpressSync/endpoints.schema.json:118]: Recommend we add a verb to the name like `RewriteRelativePaths`.  My initial reading of this I was thinking it was describing the path data rather that authorizing a new form of automation.

File: WordpressSync/index.js [hunk 3]
```diff
@@ -293,7 +257,42 @@ module.exports = async () => {
...
+
+    /**
+     * Point specific WP content to static file hosting
+     * @param {*} jsonData 
+     * @param {*} WpRow 
+     * @param {*} HTML 
+     */
+    const convertWpEndpointsToStatic = (jsonData,WpRow,HTML,type) => {
+      let apiSearchString = wordPressApiUrl;
+      if (apiSearchString.charAt(apiSearchString.length - 1) === "/") {
```
Comment (carterm) [inline WordpressSync/index.js:269]: Use `endsWith()` ?  https://www.w3schools.com/Jsref/jsref_endswith.asp

Comment (carterm) [reply WordpressSync/index.js:269]: Or for this case just remove the trailing slash with replace...`.replace(/\/$/, "")`

**Comment by jbum** [comment]
Is this still needed? Not sure where we are with it...

**Comment by carterm** [comment]
> Is this still needed? Not sure where we are with it...

This PR includes string replacement features.  Will add an issue and reference this.

**Comment by carterm** [comment]
https://github.com/cagov/wordpress-to-github/issues/11

**Comment by carterm** [comment]
https://github.com/cagov/wordpress-to-github/issues/16

**Comment by carterm** [comment]
Still not sure if the dev/prod hosts requirements are satisfied.

**Comment by chachasikes** [comment]
Out of sync now & this PR was mostly just feedback. Closing.

**Pull request closed without merging**